app/Http/Controllers/API/PageConfigController.php

```diff
@@ -7,119 +7,291 @@
+    $data['updated_at'] = gmdate('Y-m-d H:i:s');
+
+    $validator = Validator::make($data, $valRules);
+
+    if (!$validator->passes()) {
         return response()->json([
-            'pageConfig' => $pageConfig,
-            'message' => 'Success, Page Config updated successfully'
-        ], 200);
+            'success' => false,
+            'message' => 'Validation errors',
+            'errors' => $validator->errors()->all()
+        ], 422);
+    }
+
+    // Extract header_img from data
+    $headerImg = $data['header_img'];
+    unset($data['header_img']);
+
+    $pageConfig = PageConfig::findOrFail($data['id']);
+    $pageConfig->fill($data);
+    $pageConfig->save();
 
+    // Fetch existing header_img entries
+    $existingHeaderImages = FileMapper::where('ref_id', $pageConfig->id)
+        ->where('ref_type', 'page_config')
+        ->pluck('file_id')
+        ->toArray();
+
+    // Determine entries to add and remove
+    $imagesToAdd = array_diff($headerImg, $existingHeaderImages);
+    $imagesToRemove = array_diff($existingHeaderImages, $headerImg);
+
+    // Delete obsolete header_img entries if there are any
+    if (count($imagesToRemove) > 0) {
+        FileMapper::where('ref_id', $pageConfig->id)
+            ->where('ref_type', 'page_config')
+            ->whereIn('file_id', $imagesToRemove)
+            ->delete();
+    }
+
+    // Add new header_img entries if there are any
+    if (count($imagesToAdd) > 0) {
+        foreach ($imagesToAdd as $fileId) {
+            FileMapper::create([
+                'ref_id' => $pageConfig->id,
+                'ref_type' => 'page_config',
+                'file_id' => $fileId,
+                'updated_by' => $data['updated_by'],
+            ]);
+        }
     }
 
+  
+    $pageConfigArray = $pageConfig->toArray();
+    $pageConfigArray['header_img'] = $headerImg;
+
+    return response()->json([
+        'data' => $pageConfigArray,
+        'message' => 'Success, Page Config updated successfully'
+    ], 200);
+}
+
+
     public function all(Request $request)
-    {
-        $start = $request->input('start', 0);
-        $limit = $request->input('limit', 10); 
-        $tenantId = $request->header('tenant_id',0); 
-        $content = PageConfig::where([["tenant_id", $tenantId]])
-            ->offset($start)
-            ->limit($limit)
-            ->get();
-        return $content;
+{
+    
+    $tenantId = $request->header('tenant_id');
+
+    // Validate the tenant_id
+    $validator = Validator::make(['tenant_id' => $tenantId], [
+        'tenant_id' => 'required|integer|exists:tenants,id',
+    ]);
+
+    if ($validator->fails()) {
+        return response()->json([
+            'success' => false,
+            'message' => 'Validation errors',
+            'errors' => $validator->errors()
+        ], 422);
     }
+
+    $start = $request->input('start', 0);
+    $limit = $request->input('limit', 10); 
+
+    $pageConfigs = PageConfig::where('tenant_id', $tenantId)
```

Comment:
This is too expensive and slow call, use left join statement to do it with one call, thats the reason we have a mapper!